components/dashboard/src/data/current-user/update-mutation.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { User } from "@gitpod/gitpod-protocol";
+import { useMutation } from "@tanstack/react-query";
+import { getGitpodService } from "../../service/service";
+
+type UpdateCurrentUserArgs = Partial<User>;
+
+export const useUpdateCurrentUserMutation = () => {
+    return useMutation({
+        mutationFn: async (partialUser: UpdateCurrentUserArgs) => {
+            return await getGitpodService().server.updateLoggedInUser(partialUser);
+        },
+    });
+};
```

Comment:
**Suggestion (non-blocking):** How about we split it out in chunks, it can be also useful in the future to reuse the similar functionality any other place

```suggestion
export const useUpdateCurrentUserMutation = () => {
    return useMutation({
        mutationFn: updateUser,
    });
};

const updateUser = async (partialUser: UpdateCurrentUserArgs) => {
    return await getGitpodService().server.updateLoggedInUser(partialUser);
};

```

Reply:
Yah, good thinking. I'm going to hold off on extracting these for now until there's another mutation that would use the function, but that would be a good way to share some logic.